[Ukrainian] Translate `/plugins/transformer-throttler.md` (#750)
grammyjs/website: site/docs/uk/plugins/transformer-throttler.md

# Обмеження запитів до API (`transformer-throttler`)

> Цей плагін розміщує вихідні запити API в черзі через [вузьке місце](https://github.com/SGrondin/bottleneck), щоб запобігти перевищенню вашим ботом [лімітів](https://core.telegram.org/bots/faq#my-bot-is-hitting-limits-how-do-i-avoid-this), як описано в цьому [просунутому розділі](../advanced/flood.md) документації.

Reviewer: ```suggestion
Цей плагін виконує вихідні запити до API в черзі через [Bottleneck](https://github.com/SGrondin/bottleneck), щоб запобігти перевищенню вашим ботом [лімітів](https://core.telegram.org/bots/faq#my-bot-is-hitting-limits-how-do-i-avoid-this), як це описано в [просунутому розділі](../advanced/flood.md) документації.
```

run(bot);
```

</CodeGroupItem>
 <CodeGroupItem title="JavaScript">

```js
const { Bot } = require("grammy");
const { run } = require("@grammyjs/runner");
const { apiThrottler } = require("@grammyjs/transformer-throttler");

const bot = new Bot("");

const throttler = apiThrottler();
bot.api.config.use(throttler);

bot.command("example", (ctx) => ctx.reply("Я обмежений"));

// Якщо ви використовуєте цей плагін, то ви, швидше за все, захочете використати runner для паралельної обробки оновлень.
run(bot);
```

</CodeGroupItem>
 <CodeGroupItem title="Deno">

```ts
import { Bot } from "https://deno.land/x/grammy/mod.ts";
import { run } from "https://deno.land/x/grammy_runner/mod.ts";
import { apiThrottler } from "https://deno.land/x/grammy_transformer_throttler/mod.ts";

const bot = new Bot("");

const throttler = apiThrottler();
bot.api.config.use(throttler);

bot.command("example", (ctx) => ctx.reply("Я обмежений"));

// Якщо ви використовуєте цей плагін, то ви, швидше за все, захочете використати runner для паралельної обробки оновлень.
run(bot);
```

</CodeGroupItem>
</CodeGroup>

## Налаштування

> Плагін приймає один необовʼязковий аргумент наступного вигляду:

Reviewer 2: ```suggestion
Плагін приймає один необовʼязковий обʼєктний аргумент наступного вигляду:
```

```ts
type ThrottlerOptions = {
  global?: Bottleneck.ConstructorOptions; // для обмеження всіх викликів api

>   group?: Bottleneck.ConstructorOptions; // для обмеження вихідних повідомлень груп
>   out?: Bottleneck.ConstructorOptions; // для обмеження вихідних приватних повідомлень

Reviewer: ```suggestion
  group?: Bottleneck.ConstructorOptions; // для обмеження вихідних повідомлень у групах
  out?: Bottleneck.ConstructorOptions; // для обмеження вихідних приватних повідомлень
```

};
```

> Повний список властивостей обʼєкта, доступних для `Bottleneck.ConstructorOptions`, можна знайти на сторінці [вузького місця](https://github.com/SGrondin/bottleneck#constructor).

Reviewer: ```suggestion
Повний список властивостей обʼєкта, доступних для `Bottleneck.ConstructorOptions`, можна знайти на сторінці [Bottleneck](https://github.com/SGrondin/bottleneck#constructor).
```

Якщо не передано жодного аргументу, створений обмежувач використовуватиме параметри конфігурації за замовчуванням, які мають бути придатними для більшості випадків використання.
Конфігурація за замовчуванням наступна:

```ts

> // Вихідний глобальний обмежувач
> const globalConfig = {
>   reservoir: 30, // кількість нових завдань, які обмежувач прийме на старті
>   reservoirRefreshAmount: 30, // кількість завдань, які дросель прийме після оновлення
>   reservoirRefreshInterval: 1000, // інтервал у мілісекундах, з яким оновлюватиметься резервуар
> };
> 
> // Вихідний груповий обмежувач
> const groupConfig = {
>   maxConcurrent: 1, // тільки 1-е завдання одночасно
>   minTime: 1000, // чекати стільки-то мілісекунд, щоб бути готовим, після виконання завдання
>   reservoir: 20, // кількість нових завдань, які дросель прийме на старті
>   reservoirRefreshAmount: 20, // кількість завдань, які дросель прийме після оновлення
>   reservoirRefreshInterval: 60000, // інтервал у мілісекундах, з яким оновлюватиметься резервуар
> };
> 
> // Вихідний приватний обмежувач
> const outConfig = {
>   maxConcurrent: 1, // тільки 1-е завдання одночасно
>   minTime: 1000, // чекати стільки-то мілісекунд, щоб бути готовим, після виконання завдання
> };

Reviewer: ```suggestion
// Глобальний обмежувач для вихідних запитів
const globalConfig = {
  reservoir: 30, // кількість нових завдань, які обмежувач візьме після старту
  reservoirRefreshAmount: 30, // кількість завдань, які обмежувач візьме після оновлення резервуару
  reservoirRefreshInterval: 1000, // інтервал у мілісекундах, з яким оновлюватиметься резервуар
};

// Обмежувач для вихідних запитів у групах
const groupConfig = {
  maxConcurrent: 1, // тільки 1-е завдання одночасно
  minTime: 1000, // чекати стільки-то мілісекунд після виконання завдання
  reservoir: 20, // кількість нових завдань, які обмежувач візьме після старту
  reservoirRefreshAmount: 20, // кількість завдань, які обмежувач візьме після оновлення резервуару
  reservoirRefreshInterval: 60000, // інтервал у мілісекундах, з яким оновлюватиметься резервуар
};

// Обмежувач для вихідних запитів у приватних чатах
const outConfig = {
  maxConcurrent: 1, // тільки 1-е завдання одночасно
  minTime: 1000, // чекати стільки-то мілісекунд після виконання завдання
};
```
